actions/raw_action.go

```diff
@@ -93,9 +96,29 @@ func (raw *RawAction) Run(context *debos.DebosContext) error {
 		return fmt.Errorf("Failed to read %s", s)
 	}
 
-	target, err := os.OpenFile(context.Image, os.O_WRONLY, 0)
-	if err != nil {
-		return fmt.Errorf("Failed to open image file %v", err)
+	var target *os.File
+	if raw.Partition != "" {
+		devicePath := ""
+		for _, p := range context.ImagePartitions {
+			if p.Name == raw.Partition {
+				devicePath = p.DevicePath
+				break
+			}
+		}
+
+		if devicePath == "" {
+			return fmt.Errorf("Failed to find partition named %s", raw.Partition)
+		}
+
+		target, err = os.OpenFile(devicePath, os.O_WRONLY, 0)
+		if err != nil {
+			return fmt.Errorf("Failed to open partition file %v", err)
+		}
+	} else {
+		target, err = os.OpenFile(context.Image, os.O_WRONLY, 0)
```

Comment:
I think this should be rewritten a bit since it is almost the same as 'target' initialization above for `devicePath`.